pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -0,0 +1,118 @@
+package dnspolicy
+
+import (
+	"context"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crlog "sigs.k8s.io/controller-runtime/pkg/log"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/kuadrant/kuadrant-operator/pkg/reconcilers"
+
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/apis/v1alpha1"
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/traffic"
+)
+
+func (r *DNSPolicyReconciler) reconcileDNSRecords(ctx context.Context, dnsPolicy *v1alpha1.DNSPolicy, gwDiffObj *reconcilers.GatewayDiff) error {
+	log := crlog.FromContext(ctx)
+
+	for _, gw := range gwDiffObj.GatewaysWithInvalidPolicyRef {
+		log.V(1).Info("reconcileDNSRecords: gateway with invalid policy ref", "key", gw.Key())
+		//ToDo Since gateways own DNSRecords I don't think there is anything to do here?
+	}
+
+	// Reconcile DNSRecords for each gateway directly referred by the policy (existing and new)
+	for _, gw := range append(gwDiffObj.GatewaysWithValidPolicyRef, gwDiffObj.GatewaysMissingPolicyRef...) {
```

Comment:
why do we reconcile dns records when ```GatewaysMissingPolicyRef``` not quite getting that? is that the current policy being reconciled used to target another gateway and so we will apply this policy to that gateway and the new target? Would we not expect that now there is no valid policy a new default for that gateway would be created?

Reply:
This might explain it better https://github.com/Kuadrant/multicluster-gateway-controller/pull/237#discussion_r1232097666. GatewaysWithValidPolicyRef and GatewaysMissingPolicyRef both mean the policy applies. For GatewaysMissingPolicyRef, the policy ref will be added later on in the reconcile.